src/errors.ts

```diff
@@ -0,0 +1,13 @@
+export class SpOptionsEmptyError extends Error {
+    constructor(message?: string) {
+        super(message);
+        Error.captureStackTrace(this, SpOptionsEmptyError);
+    }
+ }
+
+ export class SpMultipleSourceOptionsError extends Error {
+    constructor(message?: string) {
+        super(message);
+        Error.captureStackTrace(this, SpMultipleSourceOptionsError);
+    }
+ }
```

Comment:
Curious why we have 2 errors here vs using messaging?

Reply:
`instanceof` tests? Like test for the class instead of the message content. https://jestjs.io/docs/en/expect.html#tothrowerror

Reply:
```js
expect(drinkOctopus).toThrowError(DisgustingFlavorError);
```

Reply:
lol sure - easier to test, but I think messaging is important too but its what users will see. Getting that in a standardized helps the users understand the error. Maybe we can start by organizing our error messaging?

Reply:
Agree. I think I'm going to drop these custom Errors and rely on messaging.